### DISCUSSION TITLE
How do I point the workflow .yml to a package.json in a subdirectory?

### DISCUSSION THREAD

**Discussion by suhan0694** [community/community#26196] [Actions]
<p>Say I have two packages in my repo.</p>
<ol>
<li>
<p>Client</p>
</li>
<li>
<p>Server</p>
</li>
</ol>
<p>I want an action to run only on the tests defined in the server/package.json. Can I achieve this? </p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/suhan0694">@suhan0694</a> ,</p>
<p>Do you have published the two packages to GitHub Packages registry in the repository, or you just store the source code of them in the repository?</p>
<p>Can you share your repository with us so that we can check more details about the packages and the workflow you have set up.</p>

**Comment by suhan0694** [comment]
<p>Hello <a class="mention" href="/brightran">@brightran</a> ,</p>
<p>I am not using Github package registry, just storing the source code in the repository.</p>
<p>Here is a link for the REPO <a href="https://github.com/suhan0694/mern-boilerplate" rel="nofollow noopener">https://github.com/suhan0694/mern-boilerplate</a>.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/suhan0694">@suhan0694</a> ,</p>
<p>Looks like, you define the tests in the <strong>scripts</strong>  property of the <strong>package.json</strong> file. If you want to run the scripts in the Server package, you can try the following steps:</p>
<ol>
<li>
<p>Use <a href="https://github.com/actions/checkout" rel="nofollow noopener"><strong>checkout action</strong></a> to check out the repository to the workspace (<a href="https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github.workspace</strong></a> or <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener"><strong>GITHUB_WORKSPACE</strong></a>) of the workflow.</p>
</li>
<li>
<p>Switch the  <strong>working-directory</strong> of the shell to the directory where the package.json file of Server is located, for example execute  <strong>cd server</strong>  in your case.</p>
</li>
<li>
<p>Execute the <a href="https://docs.npmjs.com/cli/run-script.html" rel="nofollow noopener"><strong>npm-run-script</strong></a> command to run the scripts in the package.json file.</p>
</li>
</ol>
<p>Hope this can help you.</p>
